src/components/general/SourcesFilter/index.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { useCallback, useMemo } from 'react';
+import {
+    _cs,
+    isDefined,
+    isNotDefined,
+} from '@togglecorp/fujs';
+import {
+    useForm,
+    ObjectSchema,
+    arrayCondition,
+} from '@togglecorp/toggle-form';
+import {
+    TextInput,
+    DateInput,
+    Button,
+    Container,
+    MultiSelectInput,
+    useBooleanState,
+} from '@the-deep/deep-ui';
+import {
+    IoSearch,
+    IoClose,
+    IoChevronUpOutline,
+    IoChevronDownOutline,
+} from 'react-icons/io5';
+
+import {
+    useRequest,
+} from '#utils/request';
+import _ts from '#ts';
+import { KeyValueElement, LeadOptions } from '#typings';
+import NonFieldError from '#components/ui/NonFieldError';
+
+import styles from './styles.scss';
+
+type FormType = {
+    status?: string[];
+    createdAt?: string;
+    publishedOn?: string;
+    assignee?: string[];
+    search?: string;
+};
+type FormSchema = ObjectSchema<FormType>;
+type FormSchemaFields = ReturnType<FormSchema['fields']>;
+
+const schema: FormSchema = {
+    fields: (): FormSchemaFields => ({
+        status: [arrayCondition],
+        createdAt: [],
+        publishedOn: [],
+        assignee: [arrayCondition],
+        search: [],
```

Comment:
Can we add all the lead filters that are possible to add right now?
Like priority, status, etc.